wiki/dfg_nodes.md

```diff
@@ -0,0 +1,53 @@
+# Understanding Data Flow Graph (DFG) Node Labels in Flowr
+
+## Overview
+The Data Flow Graph (DFG) in Flowr represents the dependencies between different code components. Each node in the DFG contains specific labels that provide information about the operation being performed, the code span, and its relationship with other nodes.
+
+This document clarifies the meaning of these labels and how to interpret them.
+
+## Node Label Structure
+Each node in the DFG follows this general format:
+
+```
+[NodeType] (NodeID) LineRange ParentNodes
+```
+
+### **Breakdown of Components:**
+- **NodeType:** Indicates the type of operation. Example: `[RBinaryOp]` for a binary operation.
+- **NodeID:** A unique identifier for the node. Example: `(7)`.
+- **LineRange:** The span of lines in the source code that the node covers. Example: `2.1-10` (meaning the node spans lines 2.1 to 10).
+- **ParentNodes:** The IDs of nodes that this node depends on. Example: `(3,6)` (indicating that this node depends on nodes 3 and 6).
```

Comment:
These are _not_ the parent nodes! In case of a [function call](https://github.com/flowr-analysis/flowr/wiki/Dataflow-Graph#3-function-call-vertex) these are the ids of the arguments, in order of occurrence.